dGame/dComponents/QuickBuildComponent.h

```diff
@@ -24,17 +24,17 @@ class QuickBuildComponent final : public Component {
 public:
 	static constexpr eReplicaComponentType ComponentType = eReplicaComponentType::QUICK_BUILD;
 
-	QuickBuildComponent(Entity* entity);
+	QuickBuildComponent(Entity* const entity);
 	~QuickBuildComponent() override;
 
 	void Serialize(RakNet::BitStream* outBitStream, bool bIsInitialUpdate) override;
-	void Update(float deltaTime) override;
+	void Update(const float deltaTime) override;
 
 	/**
 	 * Handles a OnUse event from some entity, initiating the quick build
 	 * @param originator the entity that triggered the event
 	 */
-	void OnUse(Entity* originator) override;
+	void OnUse(Entity* const originator) override;
```

Comment:
adding const here makes it not match the function its overloading, same with Update.  These not matching semantically is odd and not something I think would be good design, the overloads should match the base function it is overloading imo.  Thoughts?

Reply:
I agree. I forgot this was an overload.